torch/_dynamo/variables/tensor.py

```diff
@@ -1150,7 +1150,9 @@ def as_tensor(self, tx):
         if self._tensor_var is None:
             self._tensor_var = VariableTracker.build(
                 tx, torch.scalar_tensor
-            ).call_function(tx, [self], {})
+            ).call_function(
+                tx, [self], {"dtype": VariableTracker.build(tx, torch.float64)}
```

Comment:
You can't do it this way, right? Need to handle int and bool too

Reply:
I must have misunderstood your comment here: https://github.com/pytorch/pytorch/pull/137782/files#r1809259927

I thought you meant this code path would only be exercised by symfloats so therefore the conditional was unnecessary. Now I'm curious, what did you mean by the comment? Did you mean to enumerate all the possible dtypes?

I'll push up a new commit with the conditionals added back and add a test.

Reply:
In your old PR you specifically tested for float and only passed dtype arg when it was a float. My suggestion was that you should always compute the right dtype (whether it is float or not) and always pass it in, instead of ever relying on type inference

Reply:
Gotcha that makes sense. I've updated the PR accordingly.